meson.build

```diff
@@ -129,170 +129,207 @@ thread_dep = dependency('threads') # pthread for tensorflow-lite
 # Protobuf
 protobuf_dep = dependency('protobuf', version: '>= 3.6.1', required: false)
 
-# Orc
-have_orcc = false
-
-if get_option('enable-orc')
-  orc_dep = dependency('orc-0.4', version: '>= 0.4.17', required: true)
-  orcc = find_program('orcc', required: true)
-  if orc_dep.found() and orcc.found()
-    have_orcc = true
-    orcc_args = [orcc, '--include', 'glib.h']
-    add_project_arguments('-DHAVE_ORC=1', language: ['c', 'cpp'])
-  else
-    error('Cannot find orc library')
+## nnfw
+nnfw_dep = dependency('dummy-nnfw-dependency', required: false)
+if not get_option('nnfw-runtime-support').disabled()
+  nnfw_dep = dependency('nnfw', required: false)
+  if not nnfw_dep.found()
+    # Until nnfw supports pkg-config, we need to do this primitively.
+    nnfw_dep = cc.find_library('nnfw-dev', required: get_option('nnfw-runtime-support'))
   endif
-else
-  add_project_arguments('-DDISABLE_ORC=1', language: ['c', 'cpp'])
 endif
 
-# NO Video support
-if get_option('disable-video-support')
-  add_project_arguments('-DNO_VIDEO=1', language: ['c', 'cpp'])
-  message('Disable Video Type Support')
-endif
-
-# NO Audio support
-if get_option('disable-audio-support')
-  add_project_arguments('-DNO_AUDIO=1', language: ['c', 'cpp'])
-  message('Disable Audio Type Support')
-endif
-
-# Tensorflow
-if get_option('enable-tensorflow')
-  tf_dep = dependency('tensorflow', required: true)
+#orc
+pg_orcc = find_program('orcc', required: get_option('orcc-support'))
+orc_dep = dependency('orc-0.4', version: '>= 0.4.17', required: get_option('orcc-support'))
+
+# features registration to be controlled
+#
+# register feature as follows
+# [ <string: feature_name>, {
+#      deps: <list>, is_available: <boolean = 'all deps found when omitted'>, is_available_fallback: <boolean>,
+#      following variables handled with _when_available/_when_not_available postfix,
+#      project_args: <dict>, message: <string>, extra_args: <dict: meson variables to be registered>
+# } ]
+
+features = {
+  'video-support': {
+    'is_available': not get_option('video-support').disabled(),
+    'project_args_when_not_available': { 'NO_VIDEO': 1 },
+  },
+  'audio-support': {
+    'is_available': not get_option('audio-support').disabled(),
+    'project_args_when_not_available': { 'NO_AUDIO': 1 },
```

Comment:
Let's rename the string key: ``` s/'project_args_when_not_available'/'project_args'/ ```  We know they are used if the feature is enabled.

Reply:
Changed :)